youtube_dl/extractor/giphy.py

```diff
@@ -0,0 +1,61 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import re
+
+from .common import InfoExtractor
+from ..utils import (
+    int_or_none,
+    remove_end,
+    remove_start,
+)
+
+
+class GiphyIE(InfoExtractor):
+    _VALID_URL = r'https://(?:i|media)\.giphy\.com/media/(\w+)/|https://giphy\.com/gifs/[^/#?]+-(\w+)(?:[#?]|$)'
+    _TESTS = [{
+        'url': 'https://media.giphy.com/media/2rj8VysAig8QE/giphy.gif',
+        'md5': 'faa04a4e3134cb74005b78b28c34cc29',
+        'info_dict': {
+            'id': '2rj8VysAig8QE',
+            'ext': 'mp4',
+            'title': 'Css GIF',
+        }
+    }, {
+        'url': 'https://i.giphy.com/media/oY7zuBsNna7CM/giphy.mp4',
+        'only_matching': True
+    }, {
+        'url': 'https://giphy.com/gifs/quit-playin-MZAvNpYCrFUJi',
+        'only_matching': True
+    }]
+
+    def _real_extract(self, url):
+        [video_id] = filter(None, re.match(self._VALID_URL, url).groups())
+        page_url = 'https://media.giphy.com/media/{id}/giphy.gif'.format(id=video_id)
+        webpage = self._download_webpage(page_url, video_id)
+        title = self._og_search_title(webpage, fatal=True)
+        title = remove_end(title, ' - Find & Share on GIPHY')
+        formats = []
+        for prop in 'image', 'video':
+            url = self._og_search_property(prop, webpage)
```

Comment:
Should not be fatal.

Reply:
Yes, it should. There's no point in recovering from such fundamental breakage.

Reply:
Well, I'm not going to argue. You either fix it as requested or get PR rejected.